spark/src/main/scala/io/delta/sharing/spark/RemoteDeltaFileIndex.scala

```diff
@@ -166,20 +168,65 @@ private[sharing] case class RemoteDeltaCDFAddFileIndex(
     extends RemoteDeltaCDFFileIndexBase(
       params,
       deltaTableFiles.addFilesForCdf,
-      CDFColumnInfo.getInternalPartitonSchemaForCDFAddRemoveFile) {}
+      CDFColumnInfo.getInternalPartitonSchemaForCDFAddRemoveFile) {
+  override def listFiles(
+    partitionFilters: Seq[Expression],
+    dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
+    // Need to apply getPartitionValuesInDF to each file, to be consistent with
+    // makePartitionDirectories and partitionSchema. So that partitionFilters can be correctly
+    // applied.
+    val updatedFiles = deltaTableFiles.addFilesForCdf.map { a =>
+      AddFileForCDF(a.url, a.id, a.getPartitionValuesInDF, a.size, a.version, a.timestamp, a.stats)
```

Comment:
As discussed, perhaps add a comment in getPartitionValuesInDF that the function is idempotent, and calling it multiple times does not change its output.

Reply:
added